### PR TITLE
Make group “immortal” protect players from damage

### PR DESCRIPTION
**Pull request by Wuzzy2**
Implements #2719. 

Currently, players ignore the `immortal` group. This PR changes this.

Specifically, this does the following:

For players that have the armor group `immortal=1` set:

* Will be immune to (hopefully) all damage
* Breath will never decrease
* Hides the builtin health and breath statbars (oddly delayed, might need review)

Tested for:
* fall damage
* punch damage
* `damage_per_second` node damage
* `drowning` node damage
* `set_hp` if new hp would be lower

### CODE REVIEW

File: src/content_sao.cpp [hunk 1]
```diff
@@ -990,7 +990,8 @@ void PlayerSAO::getStaticData(std::string * result) const
 
 void PlayerSAO::step(float dtime, bool send_recommended)
 {
-	if (m_drowning_interval.step(dtime, 2.0f)) {
+	bool immortal = itemgroup_get(m_armor_groups, "immortal") == 1;
+	if (m_drowning_interval.step(dtime, 2.0f) && (!immortal)) {
```
Comment (SmallJoker) [inline src/content_sao.cpp]: Please create a new function (for example) `PlayerSAO::isVulnerable` to check against the conditions
1) whether PvP is enabled
2) `immortal` armor group is set

**Comment by ClobberXD** [comment]
About the duplicated code mentioned [here](https://github.com/minetest/minetest/pull/8266#discussion_r259574828), would it make sense to move the relevant code from `LuaEntitySAO` (which is where I assume it is) to the parent class that `PlayerSAO` inherits as well? That way, the same code can be used for both entities and players.

**Comment by Wuzzy2** [comment]
I am confused.

**Comment by ClobberXD** [comment]
See EDIT2 of https://github.com/minetest/minetest/pull/8266#discussion_r259574828

**Comment by Wuzzy2** [comment]
Sorry, @SmallJoker, your suggestion does not really make sense. If it's about UnitSAO, why should I check `enable_pvp`, which means PLAYER vs PLAYER, not UNIT vs UNIT. ;-)

**Comment by ClobberXD** [comment]
The logic behind this is that moving this code to UnitSAO results in this behaviour being applied to all ObjectRefs, players and entities alike, as UnitSAO is the base class. Not sure why SmallJoker suggested check `enable_pvp` though...

**Comment by SmallJoker** [comment]
UnitSAO: virtual function `isVulnerable`
PlayerSAO: function which checks against PvP and the immortal group
LuaEntitySAO: function which checks against the immortal group

**Comment by Wuzzy2** [comment]
Sorry, I don't know what I must do. Can someone less brain-dead than me please pick up this PR, maybe?

**Comment by ClobberXD** [comment]
I can adopt this PR (not saying I'm less brain-dead than you though :P)

**Comment by Wuzzy2** [comment]
Yay, thanks! You are my hero! :D

**Comment by paramat** [comment]
Replaced by #8682

**Pull request closed without merging**